factory/src/main/kotlin/io/qalipsis/core/factory/meters/CampaignMeterRegistryImpl.kt

```diff
@@ -305,53 +341,45 @@ internal class CampaignMeterRegistryImpl(
                 scenarioName = scenarioName,
                 stepName = stepName,
                 meterName = name,
-                tags = tags
+                tags = tags + additionalTags,
+                type = MeterType.TIMER
             )
         )
     }
 
     private fun timer(meterId: Meter.Id): Timer {
         return meters.computeIfAbsent(meterId) {
             TimerImpl(
-                MicrometerTimer.builder(meterId.meterName)
-                    .tags(meterId.tags.forMicrometer())
-                    .register(meterRegistry),
                 meterId,
-                meterReporter
+                meterReporter,
+                percentiles,
+                histogramCount
             )
         } as Timer
     }
 
-    /**
-     * Convert the provided tags to be used for Micrometer.
-     */
-    private fun Map<String, String>.forMicrometer() = additionalTags + this.map { Tag.of(it.key, it.value) }
-
-    /**
-     * Converts the provided details to a [Meter.Id].
-     */
-    private fun asId(name: String, tags: Iterable<Tag>): Meter.Id = asId(name, tags.associate { it.key to it.value })
-
-
     /**
      * Converts the provided details to a [MeterId].
      */
-    private fun asId(name: String, tags: Map<String, String>): Meter.Id = Meter.Id(
+    private fun asId(name: String, tags: Map<String, String>, type: MeterType): Meter.Id = Meter.Id(
         campaignKey = currentCampaignKey,
         scenarioName = tags["scenario"] ?: "",
```

Comment:
Considering that the scenario name and the step names are structural information, I would rather specify them as arguments, which can be nullable. Perhaps you will have to change the interface of the CampaignMeterRegistry for that...

Reply:
Let's ignore that change for the moment, there is enough to do yet.